mindsdb/integrations/handlers/plaid_handler/plaid_handler.py

```diff
@@ -0,0 +1,224 @@
+
+        self.connect()
+        if params.get('last_updated_datetime') is not None:
+            options = AccountsBalanceGetRequestOptions(
+                min_last_updated_datetime=datetime.strptime( 
+                    params.get('last_updated_datetime')
+                    )
+            )
+
+            response = self.api.accounts_balance_get(
+                AccountsBalanceGetRequest(
+                    access_token=self.access_token,
+                    options=options
+                )
+            )
+        else:
+            response = self.api.accounts_balance_get(
+                AccountsBalanceGetRequest(access_token=self.access_token)
+            )
+
+        messages = []
+        for obj in response['accounts']:
+            message_dict = {}
+            for i in obj.to_dict().keys():
+                if i.startswith('account_'):
+                    message_dict[i] = obj[i]
+                elif i == 'balances':
+                    dict_obj = obj[i].to_dict()
+                    for j in dict_obj.keys():
+                        message_dict[f'balance_{j}'] = dict_obj[j]
+                else:
+                    message_dict[f'account_{i}'] = obj[i]
+            messages.append(message_dict)
+        df = pd.DataFrame(messages)
+
+        return df
+
+    def get_transactions(self, params={}):
+        '''
+        Filters data from Plaid API's transaction endpoint and returns a DataFrame with the required information.
+        Args:
+            params (dict, optional): A dictionary of options to be passed to the Plaid API.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the filtered data.
+        '''
+
+        self.connect()
+        if params.get('start_date') and params.get('end_date'):
+            start_date = datetime.strptime(params.get('start_date'), '%Y-%m-%d').date()
+            end_date = datetime.strptime(params.get('end_date'), '%Y-%m-%d').date()
+        else:
+            raise Exception('start_date and end_date is required in format YYYY-MM-DD ')
+      
+        request = TransactionsGetRequest(
+            access_token=self.access_token,
+            start_date=start_date,
+            end_date=end_date,
+            options=TransactionsGetRequestOptions()
+        )
+
+        response = self.api.transactions_get(request)
+        transactions = parse_transaction(response['transactions'])
+
+        # Manipulate the count and offset parameters to paginate
+        # transactions and retrieve all available data
+        while len(transactions) < response['total_transactions']:
+            request = TransactionsGetRequest(
+                    access_token=self.access_token,
+                    start_date=start_date,
+                    end_date=end_date,
+                    options=TransactionsGetRequestOptions(
+                        offset=len(transactions)
+                    )
+            )
+        response = self.api.transactions_get(request)
```

Comment:
wrong indent?
looks like only last page is added to transactions

Reply:
Thanks for pointing out this